packages/ra-core/src/auth/useLogoutIfAccessDenied.ts

```diff
@@ -103,12 +103,17 @@ const useLogoutIfAccessDenied = (): LogoutIfAccessDenied => {
                     if (logoutUser) {
                         logout({}, redirectTo);
                     } else {
-                        if (redirectTo.startsWith('http')) {
-                            // absolute link (e.g. https://my.oidc.server/login)
-                            window.location.href = redirectTo;
+                        if (redirectTo != null) {
+                            if (redirectTo.startsWith('http')) {
+                                // absolute link (e.g. https://my.oidc.server/login)
+                                window.location.href = redirectTo;
+                            } else {
+                                // internal location
+                                navigate(redirectTo);
+                            }
                         } else {
-                            // internal location
-                            navigate(redirectTo);
+                            // If there is not redirect return false saying loguout is not successful
+                            return false;
```

Comment:
I wonder if we should rather return `true` here, since logout was explicitly not requested, and I'm afraid this will throw another error in this code:
https://github.com/marmelab/react-admin/blob/f67649dc7a3cfe05fa2bcef4e4743f7fa08acc5c/packages/ra-core/src/dataProvider/useDataProvider.ts#L124-L136